src/js/rtc_session.js

```diff
@@ -424,21 +401,23 @@ export class TalkingState extends RTCSessionState {
         this.transit(new DisconnectedState(this._rtcSession));
     }
     onIceStateChange(evt) {
-        this.logger.info('ICE Connection State: ', evt.currentTarget.iceConnectionState);
+        var iceState = this._rtcSession._strategy.onIceStateChange(evt, this._rtcSession._pc);
+        this.logger.info('ICE Connection State: ', iceState);
 
-        if (evt.currentTarget.iceConnectionState == ICE_CONNECTION_STATE.DISCONNECTED) {
+        if (iceState == ICE_CONNECTION_STATE.DISCONNECTED) {
```

Comment:
Have tested these paths for citrix strategy

Reply:
Yes, tested